crates/bifrost/src/loglet/mod.rs

```diff
@@ -156,21 +156,40 @@ pub type SendableLogletReadStream = Pin<Box<dyn LogletReadStream + Send>>;
 
 #[allow(dead_code)]
 pub(crate) struct LogletCommitResolver {
-    tx: oneshot::Sender<Result<LogletOffset, AppendError>>,
+    tx: Option<oneshot::Sender<Result<LogletOffset, AppendError>>>,
 }
 
 #[allow(dead_code)]
 impl LogletCommitResolver {
-    pub fn sealed(self) {
-        let _ = self.tx.send(Err(AppendError::Sealed));
+    pub fn sealed(mut self) {
+        let _ = self
+            .tx
+            .take()
+            .expect("must be set")
+            .send(Err(AppendError::Sealed));
     }
 
-    pub fn offset(self, offset: LogletOffset) {
-        let _ = self.tx.send(Ok(offset));
+    pub fn offset(mut self, offset: LogletOffset) {
+        let _ = self.tx.take().expect("must be set").send(Ok(offset));
     }
 
-    pub fn error(self, err: AppendError) {
-        let _ = self.tx.send(Err(err));
+    pub fn error(mut self, err: AppendError) {
+        let _ = self.tx.take().expect("must be set").send(Err(err));
+    }
+}
+
+#[derive(Debug, Clone, Copy, thiserror::Error)]
+#[error("Commit resolver was dropped")]
+struct CommitCancelled;
```

Comment:
The term "cancelled" often indicates a graceful user-requested cancellation of something. Abort on the other hand is used to denote an abrupt ungraceful one.

That said, I'm not sure if this error type adds enough context to the caller to understand why the commit was aborted. Do you have ideas on how either:
A) Making this error more precise to pin point why the append was aborted
B) Possibly remove the need for this error in lieu of an existing one?